--- a/Controller/Subscriber/NewAction.php
+++ b/Controller/Subscriber/NewAction.php
@@ -77,9 +77,10 @@ public function __construct(
     protected function validateEmailAvailable($email)
     {
         $websiteId = $this->_storeManager->getStore()->getWebsiteId();
-        if ($this->_customerSession->getCustomerDataObject()->getEmail() !== $email
+        if ($this->_customerSession->isLoggedIn()
+            && ($this->_customerSession->getCustomerDataObject()->getEmail() !== $email
             && !$this->customerAccountManagement->isEmailAvailable($email, $websiteId)
-        ) {
+        )) {
             throw new LocalizedException(
                 __('This email address is already assigned to another user.')
             );
