--- a/Newsletter/Controller/Subscriber/NewAction.php
+++ b/Newsletter/Controller/Subscriber/NewAction.php
@@ -61,9 +61,10 @@
     protected function validateEmailAvailable($email)
     {
         $websiteId = $this->_storeManager->getStore()->getWebsiteId();
-        if ($this->_customerSession->getCustomerDataObject()->getEmail() !== $email
+        if ($this->_customerSession->isLoggedIn()
+            && ($this->_customerSession->getCustomerDataObject()->getEmail() !== $email
             && !$this->customerAccountManagement->isEmailAvailable($email, $websiteId)
-        ) {
+        )) {
             throw new \Magento\Framework\Exception\LocalizedException(
                 __('This email address is already assigned to another user.')
             );
